### ISSUE TITLE
redir - clicking on title of own post leads to access denied error

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#8769]
### Expected behavior
When you click on the title of an own post you should not get an 'access denied' error.

### Actual behavior
I think this is a side effect of the latest changes to secure the 'redir' URLs. When I click on the title of a post I created I get a white page with an 'access denied' error. In this case it's a public post.

Looks like a 'host match' check is failing. Maybe because the 'redir' parameter is a relative URL w/o a host.

**The title is linked to this URL:** 
`https://libranet.de/redir/83?url=display/0b6b25a8-755e-2dc5-0c7b-73c519649692`

**Result:**
```
Forbidden
Access denied.

#0 /var/www/html/libranet/mod/redir.php(161): redir_check_url('https://librane...', 'display/0b6b25a...')
#1 /var/www/html/libranet/mod/redir.php(48): redir_magic(Object(Friendica\App), 83, 'display/0b6b25a...')
#2 /var/www/html/libranet/src/LegacyModule.php(90): redir_init(Object(Friendica\App))
#3 /var/www/html/libranet/src/LegacyModule.php(59): Friendica\LegacyModule::runModuleFunction('init', Array)
#4 [internal function]: Friendica\LegacyModule::init(Array)
#5 /var/www/html/libranet/src/App/Module.php(271): call_user_func(Array, Array)
#6 /var/www/html/libranet/src/App.php(571): Friendica\App\Module->run(Object(Friendica\Core\L10n), Object(Friendica\App\BaseURL), Object(Friendica\Util\Logger\StreamLogger), Array, Array)
#7 /var/www/html/libranet/index.php(42): Friendica\App->runFrontend(Object(Friendica\App\Module), Object(Friendica\App\Router), Object(Friendica\Core\PConfig\PreloadPConfig), Object(Friendica\App\Authentication), Object(Friendica\App\Page))
#8 {main}
```

**From the log:**
`2020-06-13T18:03:37Z index [ERROR]: URL check host mismatch {"contact":"https:\/\/libranet.de\/profile\/alfred","url":"display\/0b6b25a8-755e-2dc5-0c7b-73c519649692"} - {"file":"redir.php","line":199,"function":"redir_check_url","uid":"2d75d7","process_id":16129}`

Caused by #8720

### Steps to reproduce the problem
Write a post with a title.
Submit it.
Click on the title of the new post.

### Friendica version you encountered the problem
2020.06-dev

### Friendica source (git, zip)
git

### PHP version
7.3

### SQL version
10.3 MariaDB

**Comment by AlfredSK** [comment]
Error happens only with local posts. They have that relative URL (w/o) a host/domain.
Error doesn't happen with posts from remote Friendica nodes. These posts have an absolute URL.

**Works:**
`https://libranet.de/redir/573250?url=https://remote.node/display/7f2cc12aaa5fe2db31514424f8d73f078ff7211a`

**Fails:**
`https://libranet.de/redir/83?url=display/0b6b25a8-755e-2dc5-0c7b-73c519649692`

**Comment by AlfredSK** [comment]
I think the problem is here:

https://github.com/friendica/friendica/blob/e3f8a653a143de57d3e8fb705ebc541ac1ac2258/mod/redir.php#L186-L201

There needs to be an additional check for 'host is empty'. I made an adjustment locally. It is working now. But I really don't know what I'm doing - which is dangerous. :-) So, I will revert it and wait for a PHP pro to look at it.

**Comment by AlfredSK** [comment]
Works. Thanks! ♥️

Fixed by #8774
